src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+ */
+LeonardPolarQuad::LeonardPolarQuad(): Quadrature() {
+  _quad_type = LEONARD;
+}
+
+
+/**
+ * @brief Set the number of polar angles to initialize.
+ * @param num_polar the number of polar angles (2 or 3)
+ */
+void LeonardPolarQuad::setNumPolarAngles(const int num_polar) {
+
+  if (num_polar != 4 && num_polar != 6)
+    log_printf(ERROR, "Unable to set the number of polar angles to %d "
+               "for LeonardPolarQuad (4 or 6 angles)", num_polar);
+
+  Quadrature::setNumPolarAngles(num_polar);
+}
+
+
+/**
+ * @brief Routine to initialize the polar quadrature.
+ * @details This routine uses the tabulated values for the Leonard
+ *          polar angle quadrature, including the sine thetas and weights.
+ */
+void LeonardPolarQuad::initialize() {
+
+  /* Call parent class initialize routine */
+  Quadrature::initialize();
+
+  /* Allocate temporary arrays for tabulated quadrature values */
+  double* thetas = new double[_num_polar/2*_num_azim/4];
+
+  /* Tabulated values for the sine thetas and weights for the
+   * Leonard polar angle quadrature */
+  if (_num_polar == 4) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = asin(0.273658);
+      thetas[a*(_num_polar/2)+1] = asin(0.865714);
+    }
+  }
+
+  else if (_num_polar == 6) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = asin(0.099812);
+      thetas[a*(_num_polar/2)+1] = asin(0.395534);
+      thetas[a*(_num_polar/2)+2] = asin(0.891439);
+    }
+  }
+
+  /* Set the arrays of thetas and weights */
+  Quadrature::setThetas(thetas, _num_polar/2*_num_azim/4);
+
+  /* Deallocate temporary arrays */
+  delete [] thetas;
+}
+
+
+/**
+ * @brief Calculates total weights for every azimuthal/polar combination based
+ *        on the Leonard polar quadrature
+ * @param solve_3D Boolean indicating whether this is a 3D quadrature
+ */
+void LeonardPolarQuad::precomputeWeights(bool solve_3D) {
+
+  /* Allocate temporary arrays for tabulated quadrature values */
+  FP_PRECISION* weights = new FP_PRECISION[_num_polar/2*_num_azim/4];
+
+  /* Tabulated values for the sine thetas and weights for the
+   * Leonard polar angle quadrature */
+  if (_num_polar == 4) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a*(_num_polar/2)] = 0.139473 / 2.0;
+      weights[a*(_num_polar/2)+1] = 0.860527 / 2.0;
+    }
+  }
+
+  else if (_num_polar == 6) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a*(_num_polar/2)] = 0.017620 / 2.0;
+      weights[a*(_num_polar/2)+1] = 0.188561 / 2.0;
+      weights[a*(_num_polar/2)+2] = 0.793819 / 2.0;
+    }
+  }
+
+  /* Set the arrays of thetas and weights */
+  Quadrature::setPolarWeights(weights, _num_polar/2*_num_azim/4);
+
+  /* Deallocate temporary arrays */
+  delete [] weights;
+
+  Quadrature::precomputeWeights(solve_3D);
+}
+
+
+
+/**
+ * @brief Dummy constructor calls the parent constructor.
+ */
+GLPolarQuad::GLPolarQuad(): Quadrature() {
+  _quad_type = GAUSS_LEGENDRE;
```

Comment:
Perhaps this should have a default for `_num_polar`? Otherwise the `GLPolarQuad::initialize()` method below will pass without doing anything right?